src/Simulation/QCTraceSimulator.Tests/PrimitiveOperationsCounterTest.cs

```diff
@@ -120,6 +120,30 @@ void TDepthTest()
             Assert.Equal(1, sim.GetOperationMetric<TDepthOne>(MetricsNames.DepthCounter.Depth));
         }
 
+        [Fact]
+        void TDepthAndWidthTest()
+        {
+            TracerSimulator sim = null;
+            sim = RunDepthWidth(optimizeDepth: true);
+            Assert.Equal(4, sim.GetOperationMetric<TDepthWidth>("Width"));
+            Assert.Equal(2, sim.GetOperationMetric<TDepthWidth>("Depth"));
+            sim = RunDepthWidth(optimizeDepth: false);
+            Assert.Equal(3, sim.GetOperationMetric<TDepthWidth>("Width"));
+            Assert.Equal(3, sim.GetOperationMetric<TDepthWidth>("Depth"));
+        }
+
+        TracerSimulator RunDepthWidth(bool optimizeDepth) {
+            NewTracerConfiguration config = new NewTracerConfiguration();
+            config.UseDepthCounter = true;
+            config.OptimizeDepth = optimizeDepth;
+            config.TraceGateTimes = NewTraceGateTimes.ControlledZOnly;
+            TracerSimulator sim = new TracerSimulator(config);
+            QVoid res = TDepthWidth.Run(sim).Result;
```

Comment:
Since the result is insignificant, accessing `Result` is only being used for its side-effect (namely, waiting for completion of the Task). That can be made more explicit by calling `Wait`.
```suggestion
            TDepthWidth.Run(sim).Wait();
```

Reply:
Yes, this is copy-paste. I'll fix this.

Reply:
Changed to Wait() and removed next line alltogether, it was not needed.